ModernWallet/Classes/ViewController/Buy/BuyOptimizedViewController.swift

```diff
@@ -147,23 +147,19 @@ class BuyOptimizedViewController: UIViewController {
             .bind(to: assetListView.itemPicker.picker.rx.itemAttributedTitles) { (_, asset) in return NSAttributedString(string: asset.displayId, attributes: [NSForegroundColorAttributeName: #colorLiteral(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)]) }
             .disposed(by: disposeBag)
         
-        submitButton.rx.tap
-            .flatMap { [weak self] _ -> Observable<Void> in
-                guard let `self` = self else { return Observable<Void>.never() }
-                return PinViewController.presentOrderPinViewController(from: self, title: Localize("newDesign.enterPin"), isTouchIdEnabled: true)
+        let assetPairObservable = buyOptimizedViewModel.confirm.asObservable()
+            .filter { $0 }
+            .map { [buyOptimizedViewModel] _ -> (baseAsset: LWAssetModel, quotingAsset: LWAssetModel)? in
```

Comment:
Convert the closure return value to `TypeAlias` and instead of **nil** return `Observable<TypeAliasName>.empty()`